### ISSUE TITLE
k3 version 1.18.9 kube-system pods crashlooping

### ISSUE THREAD

**Issue by pw24** [k3s-io/k3s#2350]
**Environmental Info:**
K3s Version: 
k3s version v1.18.9+k3s1 (630bebf9)

Node(s) CPU architecture, OS, and Version: 
Linux raspberrypi 5.4.51-v7l+ #1333 SMP Mon Aug 10 16:51:40 BST 2020 armv7l GNU/Linux

Cluster Configuration: 
(single node master running on Rasp Pi)
$ kubectl get nodes
NAME          STATUS   ROLES    AGE     VERSION
raspberrypi   Ready    master   4d23h   v1.18.9+k3s1

**Describe the bug:**
kube-system pods aren't running, and crash-looping:

kubectl get pods -n kube-system
| NAME |    READY |  STATUS  |           RESTARTS |  AGE|
|--|--|--|--|--|
|helm-install-traefik-qr2bd        |       0/1  |   Completed   |       0   |       4d23h|
|svclb-traefik-kjbbr     |                 2/2   |  Running   |         6    |      4d23h|
|traefik-758cd5fc85-8s9qf         |        1/1  |   Running    |        0    |      18m|
|coredns-7944c66d8d-qwxjf      |           0/1  |   Running    |        0    |      17m|
|local-path-provisioner-6d59f47c7-4hvf2 |  0/1  |   CrashLoopBackOff |  7     |     18m|
|metrics-server-7566d596c8-nnzg2    |      0/1 |    CrashLoopBackOff |  7      |    18m|

**Steps To Reproduce:**

(unsure if the bug is directly from after instrall, or after starting to work with ingress config)
- Installed K3s: curl -sfL https://get.k3s.io | sh -
- run tutorial on https://www.youtube.com/watch?v=12taKl5iCpA&ab_channel=JustmeandOpensource (entailing)
  - enable traefik dashboard 
  - nginx deployment with clusterIP port 8001:80
  - setup ingress controller yaml

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: traefik-ingress-controller
  labels: 
    app: nginx
  annotations:
    kubernetes.io/ingress.class: 'traefik'
    ingress.kubernetes.io/ssl-redirect: "false"
spec:
  rules:
  - http:
      paths:
      - path: /nginx
        backend:
          serviceName: nginx
          servicePort: 8001
```
  - enable port forwarding (for exposing traefik dashboard locally on pi):  kubectl -n kube-system port-forward deployment/traefik 8080
  - enable socat port forwarding for pi to listen on port 8000 and forward to local port 8080 (to forward traefik dashboard): sudo socat tcp-l:8000,fork,reuseaddr tcp:localhost:8080

**Expected behavior:**
- Expect all of the k3s pods to be running without errors / fallbackloops
- Expect traefik dashboard to show k3s running environment + pods (as in youtube view)

**Actual behavior:**

> description under bug, with get pods table

**Additional context / logs:**

- core-dns logs
```
2020-10-06T07:14:22.69853612+01:00 stderr F Trace[1996325786]: [30.00108294s] [30.00108294s] END
2020-10-06T07:14:22.698573232+01:00 stderr F E1006 06:14:22.698347       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105: Failed to list *v1.Namespace: Get "https://10.43.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: i/o timeout
2020-10-06T07:14:22.699114302+01:00 stderr F I1006 06:14:22.699005       1 trace.go:116] Trace[1228547051]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105 (started: 2020-10-06 06:13:52.698251854 +0000 UTC m=+1490.421181657) (total time: 30.000696573s):
2020-10-06T07:14:22.699157803+01:00 stderr F Trace[1228547051]: [30.000696573s] [30.000696573s] END
2020-10-06T07:14:22.699185378+01:00 stderr F E1006 06:14:22.699034       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105: Failed to list *v1.Endpoints: Get "https://10.43.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: i/o timeout
2020-10-06T07:14:26.58877684+01:00 stdout F [INFO] plugin/ready: Still waiting on: "kubernetes"
2020-10-06T07:14:36.590064703+01:00 stdout F [INFO] plugin/ready: Still waiting on: "kubernetes"
2020-10-06T07:14:46.589131015+01:00 stdout F [INFO] plugin/ready: Still waiting on: "kubernetes"
2020-10-06T07:14:53.699091655+01:00 stderr F I1006 06:14:53.698892       1 trace.go:116] Trace[1263958076]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105 (started: 2020-10-06 06:14:23.697971067 +0000 UTC m=+1521.420900685) (total time: 30.000840897s):
2020-10-06T07:14:53.699175565+01:00 stderr F Trace[1263958076]: [30.000840897s] [30.000840897s] END
2020-10-06T07:14:53.699219158+01:00 stderr F E1006 06:14:53.698932       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105: Failed to list *v1.Service: Get "https://10.43.0.1:443/api/v1/services?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: i/o timeout
2020-10-06T07:14:53.703597721+01:00 stderr F I1006 06:14:53.703358       1 trace.go:116] Trace[884253640]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105 (started: 2020-10-06 06:14:23.702528169 +0000 UTC m=+1521.425457824) (total time: 30.000734988s):
2020-10-06T07:14:53.703656278+01:00 stderr F Trace[884253640]: [30.000734988s] [30.000734988s] END
2020-10-06T07:14:53.703682611+01:00 stderr F E1006 06:14:53.703401       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105: Failed to list *v1.Namespace: Get "https://10.43.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: i/o timeout
2020-10-06T07:14:53.703705612+01:00 stderr F I1006 06:14:53.703358       1 trace.go:116] Trace[1186657351]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105 (started: 2020-10-06 06:14:23.702521706 +0000 UTC m=+1521.425451324) (total time: 30.000759581s):
2020-10-06T07:14:53.703776836+01:00 stderr F Trace[1186657351]: [30.000759581s] [30.000759581s] END
2020-10-06T07:14:53.703798892+01:00 stderr F E1006 06:14:53.703453       1 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.4/tools/cache/reflector.go:105: Failed to list *v1.Endpoints: Get "https://10.43.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0": dial tcp 10.43.0.1:443: i/o timeout
```
- local-path-provisioner
```
2020-10-06T07:15:00.33138235+01:00 stderr F time="2020-10-06T06:15:00Z" level=fatal msg="Error starting daemon: Cannot start Provisioner: failed to get Kubernetes server version: Get https://10.43.0.1:443/version?timeout=32s: dial tcp 10.43.0.1:443: i/o timeout"
```
- metrics server (don't know if this is useful)
```
2020-10-06T07:16:35.489058878+01:00 stderr F       --kubelet-insecure-tls                                    Do not verify CA of serving certificates presented by Kubelets.  For testing purposes only.
2020-10-06T07:16:35.489140213+01:00 stderr F       --kubelet-port int                                        The port to use to connect to Kubelets. (default 10250)
2020-10-06T07:16:35.48919251+01:00 stderr F       --kubelet-preferred-address-types strings                 The priority of node address types to use when determining which address to use to connect to a particular node (default [Hostname,InternalDNS,InternalIP,ExternalDNS,ExternalIP])
2020-10-06T07:16:35.489250455+01:00 stderr F       --log-flush-frequency duration                            Maximum number of seconds between log flushes (default 5s)
2020-10-06T07:16:35.48931453+01:00 stderr F       --log_backtrace_at traceLocation                          when logging hits line file:N, emit a stack trace (default :0)
2020-10-06T07:16:35.489366475+01:00 stderr F       --log_dir string                                          If non-empty, write log files in this directory
2020-10-06T07:16:35.489455661+01:00 stderr F       --log_file string                                         If non-empty, use this log file
2020-10-06T07:16:35.489513495+01:00 stderr F       --logtostderr                                             log to standard error instead of files (default true)
2020-10-06T07:16:35.489566866+01:00 stderr F       --metric-resolution duration                              The resolution at which metrics-server will retain metrics. (default 1m0s)
2020-10-06T07:16:35.489618497+01:00 stderr F       --profiling                                               Enable profiling via web interface host:port/debug/pprof/ (default true)
2020-10-06T07:16:35.489999594+01:00 stderr F       --requestheader-allowed-names strings                     List of client certificate common names to allow to provide usernames in headers specified by --requestheader-username-headers. If empty, any client certificate validated by the authorities in --requestheader-client-ca-file is allowed.
2020-10-06T07:16:35.490103262+01:00 stderr F       --requestheader-client-ca-file string                     Root certificate bundle to use to verify client certificates on incoming requests before trusting usernames in headers specified by --requestheader-username-headers. WARNING: generally do not depend on authorization being already done for incoming requests.
2020-10-06T07:16:35.49017493+01:00 stderr F       --requestheader-extra-headers-prefix strings              List of request header prefixes to inspect. X-Remote-Extra- is suggested. (default [x-remote-extra-])
2020-10-06T07:16:35.490236782+01:00 stderr F       --requestheader-group-headers strings                     List of request headers to inspect for groups. X-Remote-Group is suggested. (default [x-remote-group])
2020-10-06T07:16:35.490288394+01:00 stderr F       --requestheader-username-headers strings                  List of request headers to inspect for usernames. X-Remote-User is common. (default [x-remote-user])
2020-10-06T07:16:35.490346006+01:00 stderr F       --secure-port int                                         The port on which to serve HTTPS with authentication and authorization.If 0, don't serve HTTPS at all. (default 443)
2020-10-06T07:16:35.490399173+01:00 stderr F       --skip_headers                                            If true, avoid header prefixes in the log messages
2020-10-06T07:16:35.490468248+01:00 stderr F       --stderrthreshold severity                                logs at or above this threshold go to stderr
2020-10-06T07:16:35.490588324+01:00 stderr F       --tls-cert-file string                                    File containing the default x509 Certificate for HTTPS. (CA cert, if any, concatenated after server cert). If HTTPS serving is enabled, and --tls-cert-file and --tls-private-key-file are not provided, a self-signed certificate and key are generated for the public address and saved to the directory specified by --cert-dir.
2020-10-06T07:16:35.490679177+01:00 stderr F       --tls-cipher-suites strings                               Comma-separated list of cipher suites for the server. If omitted, the default Go cipher suites will be use.  Possible values: TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_RC4_128_SHA,TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_RC4_128_SHA,TLS_RSA_WITH_3DES_EDE_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA256,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_RC4_128_SHA
2020-10-06T07:16:35.490763474+01:00 stderr F       --tls-min-version string                                  Minimum TLS version supported. Possible values: VersionTLS10, VersionTLS11, VersionTLS12
2020-10-06T07:16:35.490829531+01:00 stderr F       --tls-private-key-file string                             File containing the default x509 private key matching --tls-cert-file.
2020-10-06T07:16:35.490994292+01:00 stderr F       --tls-sni-cert-key namedCertKey                           A pair of x509 certificate and private key file paths, optionally suffixed with a list of domain patterns which are fully qualified domain names, possibly with prefixed wildcard segments. If no domain patterns are provided, the names of the certificate are extracted. Non-wildcard matches trump over wildcard matches, explicit domain patterns trump over extracted names. For multiple key/certificate pairs, use the --tls-sni-cert-key multiple times. Examples: "example.crt,example.key" or "foo.crt,foo.key:*.foo.com,foo.com". (default [])
2020-10-06T07:16:35.491083793+01:00 stderr F   -v, --v Level                                                 number for the log level verbosity
2020-10-06T07:16:35.491137238+01:00 stderr F       --vmodule moduleSpec                                      comma-separated list of pattern=N settings for file-filtered logging
2020-10-06T07:16:35.491184924+01:00 stderr F
2020-10-06T07:16:35.49127237+01:00 stderr F panic: Get https://10.43.0.1:443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication: dial tcp 10.43.0.1:443: i/o timeout
2020-10-06T07:16:35.491343241+01:00 stderr F
2020-10-06T07:16:35.491395038+01:00 stderr F goroutine 1 [running]:
2020-10-06T07:16:35.491450835+01:00 stderr F main.main()
2020-10-06T07:16:35.491500873+01:00 stderr F    /go/src/github.com/kubernetes-incubator/metrics-server/cmd/metrics-server/metrics-server.go:39 +0x10c
```

**Comment by pw24** [comment]
No idea how- / why this could've made 'everything work', but I basically (for the first time since installing k3s):
- copied the kube config to my computer (for remotely accessing the k3s cluster from computer)
- and allowed the cluster port (default 6443) through the firewall with ufw

After accessing the cluster remotely, via vscode kubernetes extension, I started noticing that the kube-system pods recovering to running states. First via vs code extension, and then validated with kubectl on the PI:

```
kubectl get pods -n kube-system
NAME                                     READY   STATUS      RESTARTS   AGE
helm-install-traefik-qr2bd               0/1     Completed   0          5d11h
svclb-traefik-kjbbr                      2/2     Running     8          5d11h
traefik-758cd5fc85-8s9qf                 1/1     Running     1          12h
coredns-7944c66d8d-qwxjf                 1/1     Running     1          12h
metrics-server-7566d596c8-nnzg2          1/1     Running     33         12h
local-path-provisioner-6d59f47c7-4hvf2   1/1     Running     35         12h
```
With this working, it was also possible to view the cluster with ingress from my Traefik dashboard:

![image](https://user-images.githubusercontent.com/7777405/95290971-cfc2b180-086e-11eb-988a-a17e68dcc7be.png)

No explanation why this could've recovered the pods to recovered states (don't nearly have enough of an idea what goes on in the inner-workings of k3s system pods), but this is all I saw. If someone is still able to provide some clarity on why / how this happened, that would also be great for future reference.
